tests/test_document.py

```diff
@@ -0,0 +1,80 @@
+import os
+import platform
+import tempfile
+from pathlib import Path
+
+import pytest
+
+from dangerzone.document import Document
+from dangerzone.errors import DocumentFilenameException
+
+from . import sample_doc, unreadable_pdf, unwriteable_pdf
+
+
+def test_input_sample_init(sample_doc: str) -> None:
+    Document(sample_doc)
+
+
+def test_input_sample_after(sample_doc: str) -> None:
+    d = Document()
+    d.input_filename = sample_doc
+
+
+def test_input_file_none() -> None:
+    """
+    Attempts to read a document's filename when no doc has been set
+    """
+    d = Document()
+    with pytest.raises(DocumentFilenameException) as e:
+        d.input_filename
+    assert "Input filename has not been set yet" in str(e.value)
+
+
+def test_input_file_non_existing() -> None:
+    with pytest.raises(DocumentFilenameException) as e:
+        Document("non-existing-file.pdf")
+    assert "Input file not found" in str(e.value)
+
+
+# XXX: This is not easy to test on Windows, as the file owner can always read it.
+# See also:
+# https://stackoverflow.com/questions/72528318/what-file-permissions-make-a-file-unreadable-by-owner-in-windows
+@pytest.mark.skipif(platform.system() == "Windows", reason="Unix-specific")
+def test_input_file_unreadable(unreadable_pdf):
```

Comment:
@apyrgio you're only missing here the type hints.

Reply:
Yeap, I'm on it as we speak.

Reply:
Done and force pushed.